libraries/DataFlash/DataFlash_File.cpp

```diff
@@ -133,38 +128,27 @@ void DataFlash_File::Init()
         return;
     }
 #endif
-    
-    if (_writebuf != NULL) {
-        free(_writebuf);
-        _writebuf = NULL;
-    }
 
     // determine and limit file backend buffersize
-    uint8_t bufsize = _front._params.file_bufsize;
+    uint32_t bufsize = _front._params.file_bufsize;
     if (bufsize > 64) {
-        // we currently use uint16_t for the ringbuffer.  Also,
-        // PixHawk has DMA limitaitons.
-        bufsize = 64;
+        bufsize = 64; // Also, PixHawk has DMA limitaitons.
```

Comment:
We can lose "Also," here.  And we could also correct the spelling on limitations" :-)